llvm/lib/Target/NVPTX/NVPTXSubtarget.h

```diff
@@ -72,6 +72,9 @@ class NVPTXSubtarget : public NVPTXGenSubtargetInfo {
 
   const SelectionDAGTargetInfo *getSelectionDAGInfo() const override;
 
+  bool has256BitMaskedLoadStore() const {
+    return SmVersion >= 100 && PTXVersion >= 88;
+  }
```

Comment:
Nit: I'd drop the `masked` part. We're only interested in the "256-bit" part at the moment and never actually use the masking. With "Masked" in the name the question "why do we want/need masking here?" kept popping up throughout the review. It also brings the questions like do non-masked 256-bit ld/st have different constraints? Are there non-256-bit masked loads/stores? Masking adds a new parametrization dimension that we currently do not care about.

If/when we add support for masked loads/stores, we'll also need to take the vector shape into account, so we'll probably need a more nuanced check for masking anyways. I think it makes sense to keep ld/st size check and masking support separate, even if they happen to share a common PTX/SM constraint.